examples/gno.land/p/gov/dao/dao.gno

```diff
@@ -0,0 +1,14 @@
+package dao
+
+// DAO defines the DAO abstraction
+type DAO interface {
+	// Propose adds a new proposal to the executor-based GOVDAO.
+	// Returns the generated proposal ID
+	Propose(comment string, executor Executor) (uint64, error)
+
+	// VoteOnProposal adds a vote to the given proposal ID
+	VoteOnProposal(id uint64, option VoteOption) error
```

Comment:
Agreed, I've dropped it as not required:

79be838

Reply:
How would this work, given that we want to enforce a generic `DAO` API in `r/gov/dao`? (`/r/gov/dao` doesn't have a reference to `simpledao`, but to `dao.DAO`, and the change methods for the implementation set a `dao.DAO`, not a specific implementation)

Reply:
A DAO implementation doesn't have to be limited to just this DAO interface; it can offer additional features like a member store and proposal voting mechanisms. The core DAO interface is about ensuring interoperability between DAOs through proposals—submitting them, checking their status, and executing them.

The specifics of voting and member management are up to each DAO. They can implement various interfaces like `p/demo/simpledao` or `p/gov/daoimpl`, which handle proposal management and member storage. The voting system doesn't need to conform to a standard interface.

While DAOs may evolve, with versions like `r/gov/dao/vX` or `vY` changing how voting works, the critical issue is maintaining proposal compatibility across DAOs. That's what keeps them working together effectively.

Let’s rename this package p/demo/dao instead of p/gov/dao

p/gov/dao can eventually exist when we’ll define what makes gov Dao unique, especially the tiers system; but this interface is about what is the minimal common interface for any DAO.

Reply:
Summary:
- remove VoteOnProposal
- add a GetProposal(id).GetStatus() or a GetProposalStatus(id)

Reply:
For the requested changes:

> remove VoteOnProposal

TODO

> add a GetProposal(id).GetStatus() or a GetProposalStatus(id)

The DAO interface already implements this in the form of a `PropStore` that returns proposals, whose status can be fetched as well

> Move this package location to p/demo/dao

Moved, I've kept this stashed until I got clarity that /r/gov/dao can import whatever it wants as a "DAO", and that we don't have a standardized API yet that everyone should follow:
[dba0379](https://github.com/gnolang/gno/pull/2581/commits/dba0379fc57b3bd6372b7af94c739612fa4c817c)